requirements/cross-chain-transfer.md

```diff
@@ -10,57 +10,59 @@ Various scenarios around these interactions between users and federation members
+Before the leader can process the transactions in the current block, they must first look into previous blocks and detect if there were any unprocessed transfers. It is important to note that a transfer cannot happen until all transfers from previous block have been processed successfully, this means a leader is also responsible for blocks belonging to other members in case they were offline.  
 
-Each node transfers the details of the deposit transactions made on chain A (block height, target addresses and corresponding amounts to transfer), then persists those details in its database.
-From that point, each node has enough information to create partially signed transactions and propagate them to the other members of the federation. Each node will also persist the partially signed transactions they receive and collect signatures (this is in case they become the leader of an unprocessed transfer and can immediately broadcast it), but only the leader will proceed to broadcasting a fully signed transaction.
+Each node transfers the details of the deposit transactions made on chain A (block height, transaction id, target address and corresponding amount to transfer), then persists those details in its database.
+From that point, each node has enough information to create partially signed transactions and propagate them to the other members of the federation. Each node will also persist the partially signed transactions they receive and collect signatures (this is in case they become the leader of an unprocessed transfer and can immediately broadcast it), but only the leader will proceed to broadcasting a fully signed transaction.  
 
-![The cross chain transfer is triggered after MaxReorg](../assets/cross-chain-transfers/happy-path-2.svg)
+![The cross chain transfers are triggered after MaxReorg](../assets/cross-chain-transfers/happy-path-2.svg)
 
-### The cross chain transfer appears on targeted chain
+### The cross chain transfers appear on targeted chain
 
 Once the leader has collected enough signatures, they can broadcast a fully signed transaction to the network and it will appear in the mempool of all the other nodes. This transaction will then be persisted in a future block (usually the very next one) of the chain B network.
-After that, each federation member monitoring chain B will match the transaction found on the chain by its inputs, outputs (including the OP_RETURN), and update the status of the transfer.
+> _NB: if a non-leader happens to broadcast a transaction before the leader, we should not worry about it: it means that a quorum of federation members has agreed upon this transaction, so it should be valid, and that when the leader finally submits its version to the mempool, it will be rejected because it uses the same utxo as the one already submitted by the non-leader, so we are not at risk of double spending._
```

Comment:
This makes sense. It is mostly to ensure an orderly broadcast that we specify that the leader should do it I think.